homeassistant/components/eheimdigital/config_flow.py

```diff
@@ -0,0 +1,93 @@
+"""Config flow for EHEIM Digital."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from aiohttp import ClientError
+from eheimdigital.hub import EheimDigitalHub
+import voluptuous as vol
+
+from homeassistant.components.zeroconf import ZeroconfServiceInfo
+from homeassistant.config_entries import SOURCE_USER, ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers import selector
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+
+from .const import DOMAIN
+
+CONFIG_SCHEMA = vol.Schema(
+    {vol.Required(CONF_HOST, default="eheimdigital"): selector.TextSelector()}
+)
+
+
+class EheimDigitalConfigFlowHandler(ConfigFlow, domain=DOMAIN):
+    """The EHEIM Digital config flow."""
+
+    data: dict[str, Any] = {}
+
+    async def async_step_zeroconf(
+        self, discovery_info: ZeroconfServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle zeroconf discovery."""
+        self.data[CONF_HOST] = host = discovery_info.host
+
+        self._async_abort_entries_match(self.data)
```

Comment:
I think the unique_id check is good enough

Reply:
It has the advantage of early return without needing to connect, especially in this case where only one device is advertising via zeroconf and so when it found one it only changes when the IP address changes or the user resets the devices and connect the devices to the network from scratch.